hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -879,6 +1232,173 @@ def __repr__(self):
             ", frozen" if self.frozen else "",
         )
 
+    def draw_integer(
+        self,
+        *,
+        min_value: Optional[int] = None,
+        max_value: Optional[int] = None,
+        # weights are for choosing an element index from a bounded range
+        weights: Optional[Sequence[float]] = None,
+        shrink_towards: int = 0,
+    ):
```

Comment:
I have intentionally not migrated `ConjectureData.integer_range` calls to `PrimitiveProvider.draw_integer(min_value=..., max_value=...)`, as the two have slightly different semantics. The latter biases towards endpoint values, for instance.